### ISSUE TITLE
[Bug] include_stop_str_in_output together with output_logits produces an exception

### ISSUE THREAD

**Issue by sven-h** [InternLM/lmdeploy#4241]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.

### Describe the bug

When generating text with the combination of `include_stop_str_in_output` set to `True` and requesting the logits via `output_logits='generation'` it produces the following error:
```
ERROR - async_engine.py:748 - [safe_run] exception caught: RuntimeError Boolean value of Tensor with more than one value is ambiguous
ERROR - async_engine.py:733 - [model_inst] exception caught: Boolean value of Tensor with more than one value is ambiguous
```

The whole idea behind the script is to make a simple classification by producing either 'yes' or 'no' tokens and get the logits for exactly those tokens.
Thus I would like to stop the generation when a 'yes' or 'no' token is created and keep the token in the ouput and get the logits from the stop token.

### Reproduction

To reproduce the error, I ran the following script:
```
from lmdeploy import pipeline, GenerationConfig, TurbomindEngineConfig
pipe = pipeline('internlm/internlm2_5-7b-chat',backend_config=TurbomindEngineConfig())
gen_config = GenerationConfig(
    output_logits='generation',
    stop_words = ['no', ' no'],
    max_new_tokens=200,
    temperature=0.0,
    do_sample=False,
    include_stop_str_in_output=True,
)
response = pipe("Repeat the following text: This text makes no sense but is long", gen_config=gen_config)
```

The error is:
```
ERROR - async_engine.py:748 - [safe_run] exception caught: RuntimeError Boolean value of Tensor with more than one value is ambiguous
ERROR - async_engine.py:733 - [model_inst] exception caught: Boolean value of Tensor with more than one value is ambiguous
```

and the stop token does not appear in the output anymore.
(When not requesting the logits, then the stop token appears in the text output).

The expected behaviour for me is that the stop word is in the text output and the logits are also available for the stop word (which is not the case right now).

### Environment

```Shell
sys.platform: linux
Python: 3.12.12 | packaged by Anaconda, Inc. | (main, Oct 21 2025, 20:16:04) [GCC 11.2.0]
CUDA available: True
MUSA available: False
numpy_random_seed: 2147483648
GPU 0: NVIDIA H100 NVL
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 12.9, V12.9.86
GCC: gcc (GCC) 11.5.0 20240719 (Red Hat 11.5.0-5)
PyTorch: 2.8.0+cu128
PyTorch compiling details: PyTorch built with:
  - GCC 13.3
  - C++ Version: 201703
  - Intel(R) oneAPI Math Kernel Library Version 2024.2-Product Build 20240605 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v3.7.1 (Git Hash 8d263e693366ef8db40acc569cc7d8edf644556d)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX512
  - CUDA Runtime 12.8
  - NVCC architecture flags: -gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_90,code=sm_90;-gencode;arch=compute_100,code=sm_100;-gencode;arch=compute_120,code=sm_120
  - CuDNN 91.0.2  (built against CUDA 12.9)
    - Built with CuDNN 90.8
  - Magma 2.6.1
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, COMMIT_SHA=a1cb3cc05d46d198467bebbb6e8fba50a325d4e7, CUDA_VERSION=12.8, CUDNN_VERSION=9.8.0, CXX_COMPILER=/opt/rh/gcc-toolset-13/root/usr/bin/c++, CXX_FLAGS= -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOROCTRACER -DLIBKINETO_NOXPUPTI=ON -DUSE_FBGEMM -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -DC10_NODEPRECATED -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Werror=range-loop-construct -Werror=bool-operation -Wnarrowing -Wno-missing-field-initializers -Wno-unknown-pragmas -Wno-unused-parameter -Wno-strict-overflow -Wno-strict-aliasing -Wno-stringop-overflow -Wsuggest-override -Wno-psabi -Wno-error=old-style-cast -faligned-new -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-dangling-reference -Wno-error=dangling-reference -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, TORCH_VERSION=2.8.0, USE_CUDA=ON, USE_CUDNN=ON, USE_CUSPARSELT=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_GLOO=ON, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=1, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, USE_ROCM_KERNEL_ASSERT=OFF, USE_XCCL=OFF, USE_XPU=OFF,

TorchVision: 0.23.0+cu128
LMDeploy: 0.11.1+32a1b08
transformers: 4.57.3
fastapi: 0.127.1
pydantic: 2.12.5
triton: 3.4.0
NVIDIA Topology:
        GPU0    CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X      60,156  1               N/A

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks
```

### Error traceback

```Shell
ERROR - async_engine.py:748 - [safe_run] exception caught: RuntimeError Boolean value of Tensor with more than one value is ambiguous
ERROR - async_engine.py:733 - [model_inst] exception caught: Boolean value of Tensor with more than one value is ambiguous
```

**Comment by windreamer** [comment]
@sven-h could you please try the fix in #4244 ?
